registry/storage/driver/s3-aws/s3.go

```diff
@@ -1164,28 +1116,46 @@ func (d *driver) doWalk(parentCtx context.Context, objectCount *int64, path, pre
 			})
 		}
 
-		sort.SliceStable(walkInfos, func(i, j int) bool { return walkInfos[i].FileInfoFields.Path < walkInfos[j].FileInfoFields.Path })
-
 		for _, walkInfo := range walkInfos {
```

Comment:
I would add a comment before this loop explaining that directory recursion is unnecessary under the assumption that `ListObjectsV2` returns objects in deterministically-sorted order with an explanation of how we make use of that property to skip directories where possible. It would probably be a good idea to quote from and link to this to justify that assumption: https://docs.aws.amazon.com/AmazonS3/latest/userguide/ListingKeysUsingAPIs.html

Reply:
Added 👍🏻